In [1]:
import pandas as pd

In [3]:
df = pd.read_csv("../data/raw/sms_spam_collection .csv")
df.head()

,label,message
0,ham,"Go until jurong point, crazy.. Available only ..."
1,ham,Ok lar... Joking wif u oni...
2,spam,Free entry in 2 a wkly comp to win FA Cup fina...
3,ham,U dun say so early hor... U c already then say...
4,ham,"Nah I don't think he goes to usf, he lives aro..."


In [21]:
df["label"].value_counts()

label
ham     4516
spam     653
Name: count, dtype: int64

In [5]:
df.shape


(5572, 2)

In [6]:
df.describe()

,label,message
count,5572,5572
unique,2,5169
top,ham,"Sorry, I'll call later"
freq,4825,30


In [9]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 5572 entries, 0 to 5571
Data columns (total 2 columns):
 #   Column   Non-Null Count  Dtype
---  ------   --------------  -----
 0   label    5572 non-null   str  
 1   message  5572 non-null   str  
dtypes: str(2)
memory usage: 542.9 KB


In [10]:
df.isnull().sum()

label      0
message    0
dtype: int64

In [12]:
df.nunique()

label         2
message    5169
dtype: int64

In [17]:
    df.duplicated().sum()

np.int64(403)

## Initial Findings
- Dataset contains 5571 observations
- 0 missing values
- 403 duplicated values